### DISCUSSION TITLE
Slow rendering of diagrams

### DISCUSSION THREAD

**Discussion by djeang** [structurizr/onpremises#94] [Q&A]
### Question

Hi,

I am evaluating structurizr-onpremise version 3208 using the Docker image. The rendering of simple diagram is quite slow, making the product practically not usable. That's even worse when deploying it on our openshift cluster.

Is it possible I miss a specific setting or so ? I didn't see anything related to this in documentation.

Kind Regards

**Comment by simonbrowndotje** [comment]
You'd need to provide some specific examples I'm afraid. Diagram rendering is done client-side (i.e. in your web browser, via a JavaScript library) and it can struggle with larger diagrams, but this will be unrelated to your OpenShift cluster. Automatic layout is typically applied on the server-side via Graphviz, so poor performance of this could be related to the resources you've allocated. You will also see a difference between storing workspaces on the file system vs using an S3 compatible service too.

**Reply by djeang** [reply]
Hi Simon,

I figured out that I am having this problem only when I use `theme default` in `views` section instead of specifying `styles` explicitly.
I don't have rendering issue anymore using :
```
styles {
            element "Person" {
                color #ffffff
                fontSize 22
                shape Person
            }
            ...
}
```
Is that a bug related to the usage of `theme default`  ?

**Reply by djeang** [reply]
It seems to be an issue specific to structurizr/onpremises, as `theme default` works well with  https://structurizr.com/dsl

**Reply by simonbrowndotje** [reply]
Are you using `autolayout` or the default set of views? If so, the on-premises server will be trying to load the themes when applying automatic layout ... and the delay will likely be caused by a slow network or a lack of Internet access (in which case you're seeing the TCP timeout). Do you see anything like this in the server logs?

**Reply by djeang** [reply]
Indeed, I use autolayout. 
I check the logs but I don't see any related to TCP timeout (using default loggig settings).
Anyway, it may be nice to display an error balloon, when something wrong happens.

**Reply by djeang** [reply]
To me, it would make sense to embed the default theme in the application to avoid delay. Also users don't expect the default theme changing without explicitly upgrading the app.

**Reply by simonbrowndotje** [reply]
I have added some code to ignore the pre-built themes, since they are not required for applying automatic layout -> https://github.com/structurizr/onpremises/commit/c626228b70a9d99adf1fff877b0af3ddec140bf4

This is now available in build 3244+.

**Reply by djeang** [reply]
Thank you Simon,

Works great !

**Comment by djeang** [comment]
Works great.
